### DISCUSSION TITLE
Pytorch with Cuda is not compatible with the Astra-Toolbox

### DISCUSSION THREAD

**Discussion by TimSelig** [astra-toolbox/astra-toolbox#375] [Q&A] [answered]
Hello everyone! 

I have the problem that I can´t work with pytorch-cuda and the astra-toolbox at the same time. In the following I have documented the procedure step by step up to the error message.

I´m working with Miniconda and created a new environment called "test-env" with 

          conda create --name test-env python=3.9 --no-default-packages

and activated the environment. Afterwards I installed pytorch with cuda with the command

          conda install pytorch torchvision pytorch-cuda=11.7 -c pytorch -c nvidia

If I now type the command 

          python -c "import torch; print(torch.cuda.is_available())"

I get "True", so I´m able to access cuda.
Afterwards I´m installing the astra-toolbox with the command 

          conda install -c astra-toolbox astra-toolbox

And if I´m now running the command 

          python -c "import torch"

I receive following error:

          Traceback (most recent call last):
            File "<string>", line 1, in <module>
            File "C:\ProgramData\miniconda3\envs\test-env\lib\site-packages\torch\__init__.py", line 122, in <module>
              raise err
          OSError: [WinError 127] Die angegebene Prozedur wurde nicht gefunden. Error loading "C:\ProgramData\miniconda3\envs\test-env\lib\site-packages\torch\lib\nvfuser_codegen.dll" or one of its dependencies.
            

If I´m just installing pytorch without cuda I don´t have this problem and can access the astra-toolbox as well as pytorch at the same time. But for my application I need the astra-toolbox and I also need to work with my GPU because it´s a Deep Learning application.

I would appreciate any advice or suggestions on how to solve this problem!

Best regards,
Tim Selig

**Comment by askorikov** [comment]
Hi Tim! I can't reproduce the issue on Fedora Linux, but it may be related to the fact that right now astra-toolbox and pytorch depend on different variants of CUDA Toolkit packages (from conda-forge and nvidia channels, respectively). You can try to force-remove the astra's dependency by `conda remove --force cudatoolkit` to see if it helps. I would also try to first intall astra and then pytorch in case that doesn't work.

**Reply by TimSelig** [reply, marked as answer]
Hi, thank you for your help.
Just to close this down, I could solve the issue by installing version 2.0 of the astra-toolbox with the command "conda install astra-toolbox=2.0" and everything works now.
